lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -575,6 +580,13 @@ static ResultTypeState updateResultTypeState(ValueKnowledge *tensor,
   return new_state;
 }
 
+static Type getPromotedResultType(ArrayRef<Type> scalarTypes) {
```

Comment:
Shall we add testing for scalar_scalar case here https://github.com/llvm/mlir-npcomp/blob/56c6e3676bbe1571dab0c7cc1fb5080846819438/test/Dialect/Torch/promote-types.mlir

Reply:
Oh, actually, we can do that in a different PR. We would need the `Torch_AtenAddOp` for testing this. The `Torch_AtenAddIntOp` would always have operands with the same type so won't really test the logic. Could you open an issue saying we need to add a testing for scalar-scalar type promotion?